tests/ludwig/utils/test_config_utils.py

```diff
@@ -57,3 +72,149 @@ def test_set_fixed_preprocessing_params_cache_embeddings(encoder_params: Dict[st
+
+
+@pytest.fixture()
+def llm_config_dict() -> Dict[str, Any]:
+    return {
+        MODEL_TYPE: MODEL_LLM,
+        BASE_MODEL: "HuggingFaceH4/tiny-random-LlamaForCausalLM",
+        INPUT_FEATURES: [{TYPE: TEXT, NAME: "in1"}],
+        OUTPUT_FEATURES: [{TYPE: TEXT, NAME: "out1"}],
+    }
+
+
+@pytest.fixture()
+def llm_config_object(llm_config_dict: Dict[str, Any]) -> ModelConfig:
+    return ModelConfig.from_dict(llm_config_dict)
+
+
+@pytest.fixture()
+def ecd_config_dict_llm_encoder() -> Dict[str, Any]:
+    return {
+        MODEL_TYPE: MODEL_ECD,
+        INPUT_FEATURES: [
+            {
+                TYPE: TEXT,
+                NAME: "in1",
+                ENCODER: {TYPE: MODEL_LLM, BASE_MODEL: "HuggingFaceH4/tiny-random-LlamaForCausalLM"},
+            }
+        ],
+        OUTPUT_FEATURES: [{TYPE: BINARY, NAME: "out1"}],
+    }
+
+
+@pytest.fixture()
+def ecd_config_object_llm_encoder(ecd_config_dict_llm_encoder: Dict[str, Any]) -> ModelConfig:
+    return ModelConfig.from_dict(ecd_config_dict_llm_encoder)
+
+
+@pytest.fixture()
+def ecd_config_dict_llm_encoder_multiple_features() -> Dict[str, Any]:
+    return {
+        MODEL_TYPE: MODEL_ECD,
+        INPUT_FEATURES: [
+            {TYPE: BINARY, NAME: "in1"},
+            {
+                TYPE: TEXT,
+                NAME: "in2",
+                ENCODER: {TYPE: MODEL_LLM, BASE_MODEL: "HuggingFaceH4/tiny-random-LlamaForCausalLM"},
+            },
+        ],
+        OUTPUT_FEATURES: [{TYPE: BINARY, NAME: "out1"}],
+    }
+
+
+@pytest.fixture()
+def ecd_config_object_llm_encoder_multiple_features(
+    ecd_config_dict_llm_encoder_multiple_features: Dict[str, Any]
+) -> ModelConfig:
+    return ModelConfig.from_dict(ecd_config_dict_llm_encoder_multiple_features)
+
+
+@pytest.fixture()
+def ecd_config_dict_no_llm_encoder() -> Dict[str, Any]:
+    return {
+        MODEL_TYPE: MODEL_ECD,
+        INPUT_FEATURES: [{TYPE: TEXT, NAME: "in1", ENCODER: {TYPE: "parallel_cnn"}}],
+        OUTPUT_FEATURES: [{TYPE: BINARY, NAME: "out1"}],
+    }
+
+
+@pytest.fixture()
+def ecd_config_object_no_llm_encoder(ecd_config_dict_no_llm_encoder: Dict[str, Any]) -> ModelConfig:
+    return ModelConfig.from_dict(ecd_config_dict_no_llm_encoder)
+
+
+@pytest.fixture()
+def ecd_config_dict_no_text_features() -> Dict[str, Any]:
+    return {
+        MODEL_TYPE: MODEL_ECD,
+        INPUT_FEATURES: [{TYPE: BINARY, NAME: "in1"}],
+        OUTPUT_FEATURES: [{TYPE: BINARY, NAME: "out1"}],
+    }
+
+
+@pytest.fixture()
+def ecd_config_object_no_text_features(ecd_config_dict_no_text_features: Dict[str, Any]) -> ModelConfig:
+    return ModelConfig.from_dict(ecd_config_dict_no_text_features)
+
+
+@pytest.fixture()
+def gbm_config_dict_llm_encoder() -> Dict[str, Any]:
+    return {
+        MODEL_TYPE: MODEL_GBM,
+        INPUT_FEATURES: [{TYPE: TEXT, NAME: "in1", ENCODER: {TYPE: "tf_idf"}}],
+        OUTPUT_FEATURES: [{TYPE: BINARY, NAME: "out1"}],
+    }
```

Comment:
Is this supposed to have `encoder` type `llm`? Or is it just named this was for consistency with the other fixtures?

Reply:
No, `tf_idf` is the only valid text encoder for GBM right now. This is in so that we have at least one GBM check.

Reply:
Fixed the fixture name